modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/PackageLoader.java

```diff
@@ -143,16 +147,31 @@ private BLangPackage loadPackage(PackageID pkgId, PackageEntity pkgEntity) {
             return null;
         }
 
+        String sourceName = null;
         if (pkgEntity.getKind() == PackageEntity.Kind.SOURCE) {
             pkgNode = this.sourceCompile((PackageSource) pkgEntity);
```

Comment:
All the source entries will be merged at this step and the PackageNode is created, aren't we suppose to check the package declarations for each source file inside somewhere there? ..